ilastik/applets/dataExport/dataExportGui.py

```diff
@@ -223,11 +227,51 @@ def initCentralUic(self):
 
         # Set up handlers
         self.batchOutputTableWidget.itemSelectionChanged.connect(self.handleTableSelectionChange)
+        # Handle the right-click event
+        self.batchOutputTableWidget.setContextMenuPolicy(Qt.CustomContextMenu)
+        self.batchOutputTableWidget.customContextMenuRequested.connect(self.showContextMenu)
 
         # Set up the viewer area
         self.initViewerStack()
         self.splitter.setSizes([150, 850])
 
+    def showContextMenu(self, pos):
+        """Handles right-click menu in the export file list."""
+        menu = QMenu(self)
+
+        row = self.batchOutputTableWidget.rowAt(pos.y())
+        exportPathItem = self.batchOutputTableWidget.item(row, Column.ExportLocation)
+        exportPath = exportPathItem.text() if exportPathItem else ""
+
+        if exportPath:
+            exportPath = PathComponents(exportPath).externalPath
+            exportPath = Path(exportPath).resolve() if exportPath else None
+
+        isValidPath = bool(exportPath) and exportPath.exists()
+
+        openFolderAction = menu.addAction("Open Export Folder")
+        openFolderAction.triggered.connect(lambda: self.openExportFolder(row))
+        openFolderAction.setEnabled(isValidPath)
```

Comment:
Atm this will only enable if the export file exists, but I think it would also be fine to enable it if the parent directory exists, even if there's no files in it.

Most of the logic between the two functions still looks like it could be extracted into a common function like `export_dir = self._get_export_dir(row)`. Or does `showContextMenu` need the checks but `openContextFolder` doesn't?